data_ingestion/common/di_url_file_to_gcs.py

```diff
@@ -11,26 +11,35 @@
 def local_file_path(filename):
   return '/tmp/{}'.format(filename)
 
-
 def url_file_to_gcs(url, url_params, gcs_bucket, dest_filename):
-  """Downloads a file from a url and uploads as a blob to the given GCS bucket.
+  """Attempts to download a file from a url and upload as a blob to the given GCS bucket.
+
+    url: The URL of the file to download.
+    url_params: URL parameters to be passed to requests.get().
+    gcs_bucket: Name of the GCS bucket to upload to (without gs://).
+    dest_filename: What to name the downloaded file in GCS. Include the file extension."""
+  try_urls_to_download_file_to_gcs([url], url_params, gcs_bucket, dest_filename)
+
+def try_urls_to_download_file_to_gcs(url_list, url_params, gcs_bucket, dest_filename):
+  """Attempts to download a file from a list of potential URLs and uploads as a blob to the given GCS bucket.
 
-     url: The URL of the file to download.
+     url_list: List of URLs where the file may be found.
      url_params: URL parameters to be passed to requests.get().
      gcs_bucket: Name of the GCS bucket to upload to (without gs://).
      dest_filename: What to name the downloaded file in GCS. Include the file extension."""
-  try:
-    local_path = local_file_path(dest_filename)
-    with requests.get(url, params=url_params) as response, open(local_path, 'wb') as f:
-      response.raise_for_status()
-      f.write(response.content)
-    storage_client = storage.Client()
-    bucket = storage_client.get_bucket(gcs_bucket)
-    blob = bucket.blob(dest_filename)
-    blob.upload_from_filename(local_path)
-    os.remove(local_path)
-  except requests.HTTPError as err:
-    logging.error("HTTP error for url %s: %s", url, err)
-  except google.cloud.exceptions.NotFound:
-    logging.error("GCS Bucket %s not found", gcs_bucket)
-    
\ No newline at end of file
+  for url in url_list:
+    try:
+        local_path = local_file_path(dest_filename)
+        with requests.get(url, params=url_params) as response, open(local_path, 'wb') as f:
+          response.raise_for_status()
+          f.write(response.content)
+        storage_client = storage.Client()
+        bucket = storage_client.get_bucket(gcs_bucket)
+        blob = bucket.blob(dest_filename)
+        blob.upload_from_filename(local_path)
+        os.remove(local_path)
+        break;
+    except requests.HTTPError as err:
+      logging.error("HTTP error for url %s: %s", url, err)
+    except google.cloud.exceptions.NotFound:
```

Comment:
I would move the check for NotFound outside the loop since this failure is unrelated to download so shouldn't cause a retry. Also I think it would be good to move the blob upload outside of the loop so the loop is more isolated to the retry of the download.